src/main/utils.js

```diff
@@ -0,0 +1,92 @@
+/*!
+GPII Application
+Copyright 2016 Steven Githens
+Copyright 2016-2017 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+The research leading to these results has received funding from the European Union's
+Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+"use strict";
+
+var os       = require("os");
+var fluid    = require("infusion");
+var electron = require("electron");
+
+var app = fluid.registerNamespace("gpii.app");
+
+
+/**
+ * Returns whether the underlying OS is Windows 10 or not.
+ * @return {Boolean} `true` if the underlying OS is Windows 10 or
+ * `false` otherwise.
+ */
+app.isWin10OS = function () {
+    var osRelease = os.release(),
+        delimiter = osRelease.indexOf("."),
+        majorVersion = osRelease.slice(0, delimiter);
+    return majorVersion === "10";
+};
+
+/**
+* Get the position of `Electron` `BrowserWindows`
+* @param width {Number} The current width of the window
+* @param height {Number} The current height of the window
+* @return {{x: Number, y: Number}}
+*/
+app.getWindowPosition = function (width, height) {
+    var screenSize = electron.screen.getPrimaryDisplay().workAreaSize;
+    return {
+        x: screenSize.width - width,
+        y: screenSize.height - height
+    };
+};
+
+/**
+ * Set the position of the Electorn `BrowserWindow` element.
+ * @param dialogWindow {BrowserWindow} The window which is to be positioned
+ * @param position {Object} The position where the window to be placed
+ * @param position.x {Number}
+ * @param position.y {Number}
+ */
+app.setWindowPosition = function (dialogWindow, position) {
+    dialogWindow.setPosition(position.x, position.y);
+};
+
+/**
+ * A function which capitalizes its input text. It does nothing
+ * if the provided argument is `null` or `undefined`.
+ * @param text {String} The input text.
+ * @return {String} the capitalized version of the input text.
+ */
+app.capitalize = function (text) {
+    if (fluid.isValue(text)) {
+        return text.charAt(0).toUpperCase() + text.slice(1);
+    }
+};
+
+/**
+ * Sends a message to the given Electron `BrowserWindow`
+ *
+ * @param window {Object} An Electron `BrowserWindow` object
+ * @param messageChannel {String} The channel to which the message to be sent
+ * @param message {String}
+ */
+app.notifyWindow = function (browserWindow, messageChannel, message) {
+    if (browserWindow) {
+        browserWindow.webContents.send(messageChannel, message);
+    }
+};
+
+/**
+ * Checks whether two objects are equal by using `JSON.stringify` internally.
+ * This  means that if the compared values are arrays or objects, the ordering
+ * of the elements/properties within them matters.
+ * @return {Boolean} `true` if the values are equal and `false` otherwise.
+ */
+app.equalsAsJSON = function (obj1, obj2) {
```

Comment:
Use of this utility is unstable since even the latest language standard doesn't specify that the property iteration order will be stable in all cases: https://stackoverflow.com/questions/30076219/does-es6-introduce-a-well-defined-order-of-enumeration-for-object-properties - please rewrite calls to this using fluid.model.diff instead - https://github.com/fluid-project/infusion/blob/master/src/framework/core/js/DataBinding.js#L1196